### PR TITLE
Comments Page

### PR DESCRIPTION
**Pull request by omermadmon**
Add a comments page to the webapp, where visitors can leave and view comments.

The comments are stored in a HashMap at this implementation, which means their are volatile - when the server restarts old comments are deleted.

The next assignment (will PR it in the next couple of days) is replacing this implementation with a persistent storage, and then the comments will be stored permanently.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+import com.google.gson.Gson;
+
+/** Servlet that returns some hard coded comments,
+    to be displayed on comments page as a list.*/
+@WebServlet("/comments")
+public class CommentsServlet extends HttpServlet {
+
+  // HashMap fpr storing hard coded comments.
```
Comment (OrpazFishel) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:30]: Not anymore :)

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    // Write comment from request to comments HashMap.
+    String firstName = request.getParameter("fname");
+    String lastName = request.getParameter("lname");
+    String commentText = request.getParameter("comment-text");
+
+    String name = formatName(firstName, lastName);
+    comments.put(name, commentText);
```
Comment (OrpazFishel) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:59]: Is your code allow more than one comment from the same user? And for anonymous users?

In addition, by formatting the name before the storing, users with the same last name and the same first letter of the first name will be considered as the same user. Try to do the formatting closer to the UI.

**Review by OrpazFishel — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java (hunk 2 above)
Comment (omermadmon) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:59]: Totally agree. As mentioned, next assignment focuses on implementing the comments feature using persistent storage, and these issues will be taken into consideration . 

Thanks!

Comment (omermadmon) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:59]: > Is the code allow more than one comment from the same user? And from anonymous users?
> 
> In addition, by formatting the name before the storing, users with the same last name and the same first letter of the first name will be considered as the same user. Try to do the formatting closer to the UI.

Totally agree. As mentioned, next assignment focuses on implementing the comments feature using persistent storage, and these issues will be taken into consideration .

Thanks!

**Pull request merged**